implementation-details.md

```diff
@@ -22,16 +22,26 @@ Each Solana node participates in gossip in one of two modes:
 
 ## Connecting to the cluster
 
-When connecting a node to the cluster, the node operator must specify an entrypoint. The entrypoint is the gossip address of the peer the node is going to initially connect to. The node starts up and then advertises itself to the cluster by sending messages to the peer identified by the entrypoint. The node sends:
-* two [push messages](#push-messages) containing its own [`Version`](gossip-protocol-spec.md#version) and [`NodeInstance`](gossip-protocol-spec.md#nodeinstance) 
-* a [pull request](#pull-requests) that contains the node's own [`LegacyContactInfo`](gossip-protocol-spec.md#legacycontactinfo)
+When connecting a node to the cluster, the node operator must specify an entrypoint. The entrypoint is the gossip address of the peer the node will initially connect to.
 
-The push messages are [propagated further](#propagating-push-messages-over-a-cluster) to other nodes in the cluster. Soon after that, the node should start receiving [pull responses](#pull-responses), push messages, and [pings](#ping-and-pong-messages) from other nodes. 
+Before the Gossip service is started, the node must discover the shred version and check own ports using the entrypoint's IP echo server listening on TCP endpoint on the gossip address.
+The node sends a IP echo server request with a list of UDP and TCP ports whose availability needs to be checked by the entrypoint node.
+The entrypoint node tries to establish a connection for TCP ports and send packets for UDP ports and then the entrypoint node replies back with the IP echo server response containing the public address of the node and a shred version.
```

Comment:
> The entrypoint node tries to establish a connection for TCP ports and send packets for UDP ports and then the entrypoint node replies back with the IP echo server response containing the public address of the node and a shred version.

The entrypoint node tries to establish a connection with the requesting node's TCP ports and sends packets to the UDP ports. The entrypoint node then replies back with the IP echo server response containing the public address of the requesting node and the cluster shred version.